crates/ra_proc_macro_srv/src/dylib.rs

```diff
@@ -9,43 +9,37 @@ use libloading::Library;
 use memmap::Mmap;
 use ra_proc_macro::ProcMacroKind;
 
-use std::io::Error as IoError;
-use std::io::ErrorKind as IoErrorKind;
+use std::io;
 
 const NEW_REGISTRAR_SYMBOL: &str = "_rustc_proc_macro_decls_";
 
-fn invalid_data_err(e: impl Into<Box<dyn std::error::Error + Send + Sync>>) -> IoError {
-    IoError::new(IoErrorKind::InvalidData, e)
+fn invalid_data_err(e: impl Into<Box<dyn std::error::Error + Send + Sync>>) -> io::Error {
+    io::Error::new(io::ErrorKind::InvalidData, e)
 }
 
-fn is_derive_registrar_symbol(symbol: &str) -> bool {
+fn is_derive_registrar_symbol(symbol: &&str) -> bool {
```

Comment:
```suggestion
fn is_derive_registrar_symbol(symbol: &str) -> bool {
```

Reply:
This one is ad-hoc, it does have to accept a double-reference, I think its reasonable to move it to into the function where it is used...

Reply:
Perhaps the original `|s| is_derive_registrar_symbol(s)` is better ? I don't have a strong opinion about it.

Reply:
Left as was)